cdk-infra/utils/buckets.ts

```diff
@@ -10,3 +16,149 @@ export const docsBucketNames = [
   'docs-java',
   'docs-node',
 ] as const;
+
+interface CustomBucketProps {
+  scope: Construct;
+  featureName: string;
+  env: AutoBuilderEnv;
+  bucketName: string;
+}
+export function createCustomBucket({ scope, featureName, env, bucketName }: CustomBucketProps): Bucket {
+  const stackBucketName = `${featureName}-${bucketName}-${env}`.toLowerCase();
+  const websiteRoutingRules = bucketName === 'docs-mongodb-org' ? getMongoDocsBucketRoutingRules(env) : undefined;
+
+  const bucket = new Bucket(scope, stackBucketName, {
+    removalPolicy: RemovalPolicy.DESTROY,
+    websiteRoutingRules,
+    bucketName: stackBucketName,
+    websiteIndexDocument: 'index.html',
+    websiteErrorDocument: 'docs-qa/404/index.html',
```

Comment:
Should this be `docs` rather than `docs-qa` ?

Reply:
Good point. I was working in `dotcomstg` and hardcoded this in. I actually need to create a mapping of environment to `websiteErrorDocument` like [here](https://github.com/mongodb/docs-worker-pool/blob/cd75999ac346b96bb8d96a74125fa83d9b85eb90/infrastructure/ecs-main/serverless.yml#L197).